04-cs-fundamentals/classroom/Binary.md

```diff
@@ -1,22 +1,277 @@
+
+## Converting Binary Numbers to Decimal
+
+Converting a binary number to decimal is relatively straightforward.  Multiply each digit by it's cooresponding power of 2.  Then add all the results together.
+
+1101<sub>2</sub> = 1 * 2<sup>3</sup> + 1 * 2<sup>2</sup> + 0 * 2<sup>1</sup> + 1 * 2<sup>0</sup>
+
+= 8 + 4 + 0 + 1 
+
+= 13
+
+<details>
+  <summary>What is 111<sub>2</sub> in decimal?</summary>
+  1 * 2<sup>2</sup> + 1 * 2<sup>1</sup> + 1 * 2<sup>0</sup> = 7
+</details>
+
+<details>
+  <summary>What is 1001<sub>2</sub> in decimal?</summary>
+  1 * 2<sup>3</sup> + 0 * 2<sup>2</sup> + 0 * 2<sup>1</sup> + 1 * 2<sup>0</sup> = 9
+</details>
+
+<details>
+  <summary>What is 1011<sub>2</sub> + 0111<sub>2</sub> in decimal?</summary>
+  1 * 2<sup>3</sup> + 0 * 2<sup>2</sup> + 1 * 2<sup>1</sup> + 1 * 2<sup>0</sup> = 11
+
+  0 * 2<sup>4</sup> + 1 * 2<sup>3</sup> + 1 * 2<sup>2</sup> + 1 * 2<sup>0</sup> = 7
+
+  11 + 7 = 18
+</details>
+
+### Converting From Decimal to Binary
+
+Converting the other way is a bit more tricky
+
+- Take the decimial number `num` and the current binary digit, starting at the least significant position.
+- Repeat until `num` is 0
+  - Divide the `num` by 2
+  - Place the remainder in the current digit of the binary number.
+  - `num` becomes `num / 2`
+
+For example for 13<sub>10</sub>
+
+| Step 	| `num` 	| Current binary number |
+|---	|---	|--- |
+| 0 	| 13 	|
+| 1 	| 6 	| 1
+| 2 	| 3 	| 01
+| 3 	| 1 	| 101
+| 4 	| 0 	| 1101
+
+**Exercises**
+
+<details>
+  <summary>What is 5<sub>10</sub> in binary?</summary>
 
-## Slide Deck
-+ Slide Deck used in class</br>
+| Step 	| `num` 	| Current binary number |
+|---	|---	|--- |
+| 0 	| 5 	|
+| 1 	| 2 	| 1
+| 2 	| 1 	| 01
+| 3 	| 0 	| 101
+
+</details>
+
+<details>
+  <summary>What is 14<sub>10</sub> in binary?</summary>
+
+| Step 	| `num` 	| Current binary number |
+|---	|---	|--- |
+| 0 	| 14 	|
+| 1 	| 7 	| 0
+| 2 	| 3 	| 10
+| 3 	| 1 	| 110
+| 4 	| 0 	| 1110
+
+</details>
+
+## ASCII & Unicode
+
+If everything in a computer is a binary number, how does a computer handle text?  Enter ASCII & Unicode.  [ASCII or the American Standard Code for Information Exchange](https://en.wikipedia.org/wiki/ASCII) was published in 1963 and is a table with text characters represented by a number.  In the early days of comptuters when the sizes of variables were small, there were a very limited number of possible characters which could be represented resulting in this table.
+
+![ASCII Table](images/ascii-table.png)
+
+Originally these were the standard characters a computer could represent.  The table was limited in size because characters were assumed to be 1 byte in length.  Today ASCII has largely been replaced by [Unicode](https://home.unicode.org/) a which much larger set of characters and multiple methods of encoding which allows a variety of languages and even emojis.  ASCII however lives on as a subset of Unicode & in [wonderful ASCII Art](https://www.asciiart.eu/).
```

Comment:
emojis -> emoji (it's already plural)

Also, I've added a line about different encodings.

```suggestion
Originally these were the standard characters a computer could represent.  The table was limited in size because characters were assumed to be 1 byte in length.  Because ASCII only really supported English text different countries came up with a variety of different encodings.

Today ASCII and other encodings have largely been replaced by [Unicode](https://home.unicode.org/)(**uni**versal **code**) a which much larger set of characters and multiple methods of encoding which allows a variety of languages and even emoji.  ASCII however lives on as a subset of Unicode & in [wonderful ASCII Art](https://www.asciiart.eu/).
```